src/proxy/pending.rs

```diff
@@ -0,0 +1,73 @@
+extern crate linkerd2_router as rt;
+
+use futures::{Future, Poll};
+
+use svc::{self, ServiceExt};
+
+type Error = Box<dyn std::error::Error + Send + Sync>;
+
+#[derive(Clone, Debug)]
+pub struct MakePending<M> {
+    inner: M,
+}
+
+/// Creates a `Service` immediately, even while the future making the service
+/// is still pending.
+pub enum Pending<F, S> {
```

Comment:
TIOLI: it seems like "pending" is a little ambiguous  --- one might assume it has something to do with tracking pending requests, for instance, especially since it's typically paired with a buffer... Would this be more accurately named `make_service_as_service` or something? That names probably no better. I think this is what finagle calls FactoryToService, fwiw...

Reply:
`FactoryToService` creates a `Service[Req, Rep]` where every `Req` will acquire a new service from the factory. I believe this is more akin to [`Service.pending`](http://twitter.github.io/finagle/docs/com/twitter/finagle/Service$.html#pending[Req,Rep](svc:com.twitter.util.Future[com.twitter.finagle.Service[Req,Rep]]):com.twitter.finagle.Service[Req,Rep]).